### DISCUSSION TITLE
proj.addmetashape empty dictionary

### DISCUSSION THREAD

**Discussion by UQDannySmith** [UTokyo-FieldPhenomics-Lab/EasyIDP#50] [Q&A] [answered]
Hi I am trying to get the quick start examples working but get an error when I try to import the metashape project files:
```
proj = idp.ProjectPool()
proj.add_metashape(["C:\\Users\\xxxx\\AppData\\Local\\easyidp.data\\2017_tanashi_lotus\\170531.Lotus.psx"])

```
this returns an empty dictionary. Could you suggest what I might be doing wrong? Thanks

**Comment by HowcanoeWang** [comment, marked as answer]
Please refer to this discussion: https://github.com/UTokyo-FieldPhenomics-Lab/EasyIDP/discussions/44#discussioncomment-3545289

And starting by the official demo: https://easyidp.readthedocs.io/en/latest/jupyter/quick_start.html

**Reply by UQDannySmith** [reply]
Thanks!

**Comment by UQDannySmith** [comment]
HI @HowcanoeWang I have managed to get the reverse calculation working with my Metashape imagery and the ROI from my shapefile. What would be the best way to export my reverse calculated images (.jpg) so that I can process them with Easy PCC? Would it be better to rotate the images, or add a mask based on the ROI? E.g here is an example plot which I've added a black mask to. Would this type of image be ok to process with Easy PCC? Thanks again for your help

![image](https://user-images.githubusercontent.com/100820490/190399832-300fc823-7f01-48d9-8ead-c96de30aa0b4.png)

**Reply by HowcanoeWang** [reply]
It is recommended to add the ROI mask to the fourth alpha layer (RGB-A) and save it into a png file, then you can eliminate the effects of black pixels on the boundary.

In the future, the EasyIDP will provide a function to save results to png files directly.

**Reply by HowcanoeWang** [reply]
In 7210b10, we have implemented this function into EasyIDP, you can save both json and png file for the output of `back2raw`, by specifying the `save_folder`:

```python
>>> img_dict_sort = roi.back2raw(ms, ..., save_folder="folder_to_save")
>>> img_dict_sort = ms.sort_img_by_distance(..., save_folder="folder_to_save")
```

Hoping this update helps you!
